grass7/temporal/t.vi/t.vi.py

```diff
@@ -0,0 +1,590 @@
+        list_strds["blue"] = blue
+        list_strds["green"] = green
+
+    if viname == "gvi" and (
+        not red or not nir or not blue or not green or not band5 or not band7
+    ):
+        gscript.fatal(
+            _("gvi index requires blue, green, red, nir, chan5 and chan7 maps")
+        )
+    elif viname == "gvi":
+        check_map_numbers(where, red=red, nir=nir)
+        list_strds.extend([red, green, blue, nir, band5, band7])
+        list_strds["red"] = red
+        list_strds["nir"] = nir
+        list_strds["blue"] = blue
+        list_strds["green"] = green
+        list_strds["band5"] = band5
+        list_strds["band7"] = band7
+
+    if viname == "ndwi":
+        from_std = green
+    else:
+        from_std = red
+    maps = from_std.get_registered_maps_as_objects(where, "start_time", None)
+
+    if not strdsout:
+        strdsout = tgis.check_new_stds(
+            options["output"], "strds", dbif=dbif, overwrite=False
+        )
+        ttype, stype, title, descr = from_std.get_initial_values()
+        title = "{vi} from {se}".format(vi=viname, se=from_std.get_name())
+        descr = "{vi} from {se}".format(vi=viname, se=from_std.get_name())
+        if where:
+            descr += " for {wh}".format(wh=where.replace("'", ""))
+        print(title, descr)
+        strdsout = tgis.open_new_stds(
+            options["output"], "strds", ttype, title, descr, stype, dbif, overwrite
+        )
+
+    times = []
+    for ma in maps:
+        if ma.get_temporal_type() == "absolute":
+            times.append(ma.get_absolute_time())
+        else:
+            times.append(ma.get_relative_time())
+
+    process_queue = pymod.ParallelModuleQueue(int(nprocs))
+    vtorast = pymod.Module("v.to.rast", memory=memory, quiet=True, run_=False)
+    vtorast.inputs.use = "val"
+    ivi = pymod.Module("i.vi", quiet=True, run_=False)
+    ivi.inputs.viname = viname
+    outlist = []
+
+    for ti in times:
+        if ti[1]:
+            mywhere = "start_time >= '{}' AND start_time <= '{}'".format(ti[0], ti[1])
+        else:
+            mywhere = "start_time = '{}'".format(ti[0])
+        rasters = {}
+        rasters_cloud = {}
+        rasters_shadow = {}
+        for k, strds in list_strds.items():
+            rast = strds.get_registered_maps_as_objects(mywhere, "start_time", None)
+            rasters[k] = rast
+        cmd_list = []
+        # control if cloud and/or shadow exists as raster or vector
+        cloud_exists = False
+        shadow_exists = False
+        extent = rast[0].get_temporal_extent()
+        clean = []
+        if clouds:
+            maps = clouds.get_registered_maps_as_objects(mywhere, "start_time", None)
+            if len(maps) == 0:
+                gscript.warning(_("No clouds map for {}").format(ti[0]))
+            else:
+                cloud_exists = True
+                if len(maps) > 1:
+                    gscript.warning(_("To many clouds maps, only first will be used"))
```

Comment:
IIUC, one single cloud map will be used to mask the full time series here, is that right? 

Maybe better warn that this is not a good idea and it might lead to unnecessary loss of valid data? Same for shadows below.

Reply:
no, only the first map for the considered times (since we are looping for each times) is used. it is just a check that no more than one map of each input map is used. I think we can resolve this conversation, if it is fine for you. Maybe I can add a comment here

Reply:
ah, ok, thanks for the explanation and sorry for my confusion. The message could then be: "Too many clouds maps for a single date/time, only the first one will be used". What do you think?

Still I wonder in which case would someone register many cloud maps for a single date/time. Maybe, when they try with different thresholds? or different flags? 

Please, add all these details to the manual.